buildenv/jenkins/jobs/infrastructure/signedOffByCheck

```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+def FAIL = false
+def SRC_REPO = 'https://github.com/${ghprbGhRepository}.git'
+def BAD_FILES = []
+def HASHES = '###################################'
+def WHITELIST_FILES = []
+def REGEXLIST = []
+def COMMITS = []
+def white = []
+def BAD_COMMITS = []
+
+timeout(time: 6, unit: 'HOURS') {
+    stage('Copyright Check') {
+        node ('worker') {
+            timestamps {
+                git url: SRC_REPO
+                sh "git fetch --tags --progress origin +refs/pull/*:refs/remotes/origin/pr/*"
+                sh "git checkout --detach ${ghprbActualCommit}"
+                sh 'git fetch origin'
+                COMMITS = sh(
+                     script : "git log origin/${ghprbTargetBranch}..HEAD",
+                     returnStdout: true
+                     ).trim()
+                echo COMMITS
+                if (COMMITS == "") {
+                    echo "There are no commits to check for signoffby message"
+                }else{
+                    def str = COMMITS.split("(?=commit [a-f0-9]{40}+\n(Merge: [a-f0-9 ]+)*\n*Author: .*\nDate:   .*+\n)");
+                    str.each { string ->
```

Comment:
Can we use better variable names other than str and string?
perhaps
`COMMITS_RAW`
`COMMITS`
`COMMIT`
?

Reply:
fixed